designs/code-generate-image-plugin.md

```diff
@@ -0,0 +1,296 @@
+			Selector: &metav1.LabelSelector{
+				MatchLabels: ls,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: ls,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image:   imageFor{{ resource }}(m.Name),
+						Name:    {{ resource }},
+                        ImagePullPolicy: {{ resource }}.Spec.ContainerImagePullPolicy,
+						Command: []string{"{{ resource }}"},
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: {{ resource }}.Spec.ContainerPort,
+							Name:          "{{ resource }}",
+						}},
+					}},
+				},
+			},
+		},
+	}
+	// Set {{ resource }} instance as the owner and controller
+	ctrl.SetControllerReference(m, dep, r.Scheme)
+	return dep
+}
+
+// labelsFor{{ resource }} returns the labels for selecting the resources
+// belonging to the given {{ resource }} CR name.
+func labelsFor{{ resource }}(name string) map[string]string {
+	return map[string]string{"type": "{{ resource }}", "{{ resource }}_cr": name}
+}
+
+// imageFor{{ resource }} returns the image for the resources
+// belonging to the given {{ resource }} CR name.
+func imageFor{{ resource }}(name string) string {
+	// TODO: this method will return the value of the envvar create to store the image:tag informed 
+}
+
+func (r *{{ resource }}Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&cachev1alpha1.{{ resource }}{}).
+		Owns(&appsv1.Deployment{}).
+		Complete(r)
+}
+
+``` 
+
+**Example of the spec for the <kind>_types.go template**
+
+```go
+// {{ resource }}Spec defines the desired state of {{ resource }}
+type {{ resource }}Spec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+    // +kubebuilder:validation:Minimum=1
+	// Size defines the number of {{ resource }} instances
+	Size int32 `json:"size,omitempty"`
+
+    // ImagePullPolicy defines the policy to pull the container images
+	ImagePullPolicy string `json:"image-pull-policy,omitempty"`
+
+    // ContainerPort specifies the port which will be used by the image container 
+	ContainerPort int `json:"container-port,omitempty"`
+
+}
+```
+
+## Design Details
+
+### Test Plan
+
+To ensure this implementation a new project example should be generated in the [testdata](../testdata/) directory of the project. See the [test/testadata/generate.sh](../test/testadata/generate.sh). Also, we should use this scaffold in the [integration tests](../test/e2e/) to ensure that the data scaffolded with works on the cluster as expected.
+
+### Graduation Criteria
+
+- The new plugin will only be support `project-version=3` 
+- The attribute image with the value informed should be added to the resources model in the PROJECT file to let the tool know that the Resource get done with the common basic code implementation: 
+
+```yaml
+plugins:
+    deploy-image.go.kubebuilder.io/v1beta1:
```

Comment:
This configuration implies that phase 2 is required to implement this plugin, since the 1.5 proposal doesn't discuss adding plugin configs to `plugins`. @Adirio that's correct right?

Reply:
project-version v3 already supports that field. Nothing specific was defined for plugin phase 1.5 but it doesn't prevent it either. So I would say it doesn't require phase 2 (supporting external binaries is something we still haven't properly designed AFAIK) but it definetely requires phase 1.5.